R/stock_time.R

```diff
@@ -1,111 +1,94 @@
 g3s_time_convert <- function (year, step = NULL) {
-    if (is.null(step)) {
-        as.integer(year)
-    } else if (any(step > 9)) {
-        as.integer(year) * 100L + as.integer(step)
-    } else {
-        as.integer(year) * 10L + as.integer(step)
-    }
-}
-
-g3s_time_multiplier <- function (times) {
-    if (all(times > 100000)) {
-        # Year + 2-char step
-        100L
-    } else if (all(times > 10000)) {
-        # Year + step
-        10L
-    } else {
-        # Just year
-        1L
-    }
+  if (is.null(step)) {
+    as.integer(year) * 100L
+  } else {
+    as.integer(year) * 100L + as.integer(step)
+  }
 }
 
 g3s_time_labels <- function (times) {
-    mult <- g3s_time_multiplier(times)
-    if (mult > 1L) {
-        sprintf("%d-%02d",
-            times %/% mult,
-            times %% mult)
-    } else {
-        as.character(times)
-    }
+  ## Steps are included
+  if (!all((times / 100) %% 1 == 0)) {
```

Comment:
Minor nitpick: ``!all()`` in the general case is dramatically slower than ``any()``, as the former will have to iterate over everything regardless, but ``any`` can stop as soon as it finds something. How true this is with R's vectorisation I'm not sure, but it's a good habit.

But more importantly I'm not convinced by the inner part. You're roughly asking "does ``times / 100`` have a fractional part", which as well as being slower by invoking floats, there's a chance the equality won't do what you want. One should also heed the warning in the ``%%`` man page.

Anyway, ``any(times %% 100 > 0)`` is what you really want here IMO. The modulo throws away everything > 100, if anything is > 0 then there's steps.

Reply:
Thanks, updated : )